### ISSUE TITLE
checking go sdk messes up the ability to build

### ISSUE THREAD

**Issue by gedw99** [go-lang-plugin-org/go-lang-idea-plugin#1705]
/Users/apple/go1.4/bin/go build -o "/private/var/folders/wp/ff6sz9qs6g71jnm12nj2kbyw0000gp/T/Build server.go and run0go" /Users/apple/Workspace/src/github.com/grpc/grpc-go/examples/route_guide/server/server.go
# google.golang.org/grpc/examples/route_guide/proto
../../../google.golang.org/grpc/examples/route_guide/proto/route_guide.pb.go:20: import /Users/apple/Workspace/pkg/darwin_amd64/github.com/golang/protobuf/proto.a: object is [darwin amd64 devel +1defd22 Tue May 26 16:50:35 2015 +0000 X:none] expected [darwin amd64 go1.4.2 X:precisestack]

Basically i was pointing to go 1.5, and then changed to point to go 1.4, but the old ".a" files are hanging around.

**Comment by dlsniper** [comment]
Short answer: Add a "-a" switch to the go tool parameters in the run configuration and it will be fixed.
Long answer:
Which plugin version do you use?
What do you mean by checking go sdk?
It's not clear what you attempt to do there. You have the IDE pointed at a symlink and then you changed the symlink target? Sorry but the description needs more :heart:

**Comment by gedw99** [comment]
Sorry:
OS: OSX 
Versions of intelliJ: 14.14
GO pluging version: 0.9.348

----

to be honest. i just started to use the plugin, and am not sure i am doing things correctly. It could be user induced bug :)

I think where i am getting messed up is on the "Run Kind" of the Configuration.

For example:

This works fine for me. Its a single file that i load as a project
Workspace/src/github.com/go-gl/examples/glfw31-gl41core-cube/cube.go


But this does not.
working dir: /Users/apple/Workspace/src/github.com/grpc/grpc-go/
file: /Users/apple/Workspace/src/github.com/grpc/grpc-go/examples/route_guide/server/server.go

It stalls with: 
"/Users/apple/go1.4/bin/go build -o "/private/var/folders/wp/ff6sz9qs6g71jnm12nj2kbyw0000gp/T/Build server.go and run7go" /Users/apple/Workspace/src/github.com/grpc/grpc-go/examples/route_guide/server/server.go
# google.golang.org/grpc/transport
../../../google.golang.org/grpc/transport/http2_client.go:48: import /Users/apple/Workspace/pkg/darwin_amd64/google.golang.org/grpc/codes.a: object is [darwin amd64 devel +1defd22 Tue May 26 16:50:35 2015 +0000 X:none] expected [darwin amd64 go1.4.2 X:precisestack]
"

**Comment by dlsniper** [comment]
Ok, so you basically have something like this:

- from the command line this works:
```bash
go build /Users/apple/Workspace/src/github.com/grpc/grpc-go/examples/route_guide/server/server.go
```

- from the IDE:
your run configuration for a Go Application configured with a File target to ```/Users/apple/Workspace/src/github.com/grpc/grpc-go/examples/route_guide/server/server.go``` does not. 

Is that correct?

Could you provide screenshots (or even better a screencast) of the problem? I guess that would help more to understand the issue.

Thanks.

**Comment by joeblew99** [comment]
Will have another crack at it on Monday, and send screenshots.

My aim to to get debugging going.

If we work up some notes and screenshots we can make it  easy for other newbies too

**Comment by dlsniper** [comment]
@joeblew99 if you refer to the GDB debugging, there's a reason debugging is not documented yet. It's experimental, even the Go manual says that GDB doesn't play nicely with it. But sure, please send screenshots and I'll be happy to help you out ;) Have a nice weekend.

**Comment by joeblew99** [comment]
Ah. Maybe I miss understand.
I assumed delve was part of the plugin.
It's not ? Its on the roadmap ?

**Comment by joeblew99** [comment]
I also thought delve does not use GDB. Correct ??

**Comment by dlsniper** [comment]
@joeblew99 I'm not sure where you've got the information but it's wrong. Delve support is on the roadmap. The only debug runner in the plugin now is GDB (and that's experimental). If you want to track the progress for it, here it is: https://github.com/go-lang-plugin-org/go-lang-idea-plugin/pull/1640

And yes, delve doesn't use GDB.

**Comment by joeblew99** [comment]
OK thanks.

**Comment by dlsniper** [comment]
@gedw99 / @joeblew99 (as I don't know if it's the same person) Any news on this?

Changing Go SDK version would cause compilation issues as well outside IDEA, basically one would need to either wipe $GOPATH/pkg dir or to use ```-a``` flag.

@zolotov what do you think, should we threat this case somehow in the IDE? Maybe display a message informing the user about this issue and how to solve it?

**Comment by zolotov** [comment]
@dlsniper maybe, but not on changing SDK. Do you know how to detect situations when GOPATH/pkg content doesn't correspond to some SDK?

**Comment by gedw99** [comment]
All - Yes i was wondering the same. 
I was changing between go 1.4 and go 1.5
Both were on my system in different directories, and both worked correctly when i setup the env variable correctly for each. 

SO the key question is, DOES the plugin rely on the environment variable ??

**Comment by dlsniper** [comment]
> SO the key question is, DOES the plugin rely on the environment variable ??

@gedw99 on what environment variable do you refer to? And where you restarting the IDE after the change?

> Do you know how to detect situations when GOPATH/pkg content doesn't correspond to some SDK?

@zolotov for this the plugin should read the .a files and see to which version they belong to but that's not fun to do because it might bump into a file that's compiled with 1.4 and another one with 1.5 and another with 1.4.2. In this case the decision should be to wipe the dir contents or display a message to the user. More over, go compiler can compile directly from .a files,  see http://stackoverflow.com/a/26524210/951414, which means that we can't go and wipe out $GOPATH/pkg to our liking. That's why I think a simple message could be enough.

**Comment by zolotov** [comment]
> That's why I think a simple message could be enough.

Message is ok but we shouldn't show it on sdk changing, there are a lot of cases when this situation might happen.

**Comment by zolotov** [comment]
> SO the key question is, DOES the plugin rely on the environment variable ??

@gedw99 It's not the key question. Plugin uses your environment by default, but of course you should restart an application after changing it, that's just how environment works. In common case you can ignore ENV settings and set GOPATH manually at `Setting | Languages & Frameworks | Go | Go Libraries`

**Comment by dlsniper** [comment]
I see. Would it be possible to store the go version alongside with the other project settings and compare it with yhe current sdk version only when run configurations are invoked? If so, I can work on something for that.

**Comment by zolotov** [comment]
> compare it with the current sdk version only when run configurations are invoked

Sounds good.

> Would it be possible (desired) to store the go version alongside with the other project settings 
It shouldn't be a project setting and shouldn't be stored in .idea. It's better to use `PropertiesComponent` just like for isNotificationShownFlags. Also likely it should be recalculated on file update events under $GOPATH/pkg directories (VirtualFileListener).
